tests/rados/stretch_cluster.py

```diff
@@ -510,3 +598,46 @@ def wait_for_alert(node: CephAdmin, alert: str, duration: int):
         f"The alert {alert} still active on cluster after timeout of {duration} seconds"
     )
     return False
+
+
+def stretch_mode_force_recovery(node: CephAdmin) -> bool:
+    """
+    When in stretch degraded mode, This module will force the cluster to go into “recovery” mode automatically
+     when the disconnected data center comes back.
+    Args:
+        node: Cephadm node where the commands need to be executed
+
+    Returns: True -> pass ( Recovery command was executed successfully)
+            False -> fail ( Recovery command could not be executed )
+    """
+    cmd = "ceph osd force_recovery_stretch_mode --yes-i-really-mean-it"
+    try:
+        node.shell([cmd])
+    except Exception:
+        log.error("Recovery with stretch mode could not be triggered")
+        return False
+    log.info("Triggered the recovery in stretch mode")
+    return True
```

Comment:
Suggestion

I believe we can extend either `run_ceph_command` or `shell` to handle exceptions gracefully by passing an argument. This way we can avoid doing the above steps below...

Reply:
Handled the same with ` run_ceph_command ` method